include/geode/geosciences_io/model/private/geos_output.h

```diff
@@ -0,0 +1,194 @@
+            region_id += 1;
+        }
+        return region_map_id;
+    }
+
+} // namespace
+
+namespace geode
+{
+    namespace detail
+    {
+        template < typename Model >
+        class GeosOutputImpl
+        {
+        public:
+            GeosOutputImpl(
+                absl::string_view files_directory, const Model& model )
+                : model_( model ),
+                  files_directory_{ ghc::filesystem::path{
+                      geode::to_string( files_directory ) }
+                                        .string() },
+                  prefix_{ geode::filename_without_extension(
+                      files_directory ) }
+            {
+                if( ghc::filesystem::path{ geode::to_string( files_directory ) }
+                        .is_relative() )
+                {
+                    ghc::filesystem::create_directory(
+                        ghc::filesystem::current_path() / files_directory_ );
+                }
+                else
+                {
+                    ghc::filesystem::create_directory( files_directory_ );
+                }
+            }
+            absl::string_view files_directory() const
+            {
+                return files_directory_;
+            }
+            absl::string_view prefix() const
+            {
+                return prefix_;
+            }
+            void write_file()
+            {
+                std::string REGION_ID_ATTRIBUTE_NAME{ "attribute" };
+                std::unique_ptr< geode::SurfaceMesh3D > surface;
+                std::unique_ptr< geode::SolidMesh3D > solid;
+                std::tie( surface, solid ) =
+                    geode::convert_brep_into_surface_and_solid( model_ );
+
+                const auto brep_component_uuid_attribute =
+                    solid->polyhedron_attribute_manager()
+                        .find_attribute<
+                            geode::uuid_from_conversion_attribute_type >(
+                            geode::uuid_from_conversion_attribute_name );
+
+                std::shared_ptr< geode::VariableAttribute< geode::index_t > >
+                    region_attribute{
+                        solid->polyhedron_attribute_manager()
+                            .template find_or_create_attribute<
+                                geode::VariableAttribute, geode::index_t >(
+                                REGION_ID_ATTRIBUTE_NAME, geode::NO_ID )
+                    };
+                auto region_map_id = create_region_attribute_map( model_ );
+
+                for( const auto polyhedron_id :
+                    geode::Range( solid->nb_polyhedra() ) )
+                {
+                    region_attribute->set_value( polyhedron_id,
+                        region_map_id
+                            .find( brep_component_uuid_attribute->value(
+                                polyhedron_id ) )
+                            ->second );
+                }
+
```

Comment:
This will not be enough I also need to convert every other attribut to get rock properties (e.g. porosity and permeability.) Do you have any idea? how to do that.